src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -415,6 +499,69 @@ private static async Task<Document> AddNullCheckStatementAsync(
         return document.WithSyntaxRoot(newRoot);
     }
 
+    private static async Task<Document> AddEnumIsDefinedCheckStatementAsync(
+        Document document,
+        IParameterSymbol parameter,
+        SyntaxNode functionDeclaration,
+        IMethodSymbol method,
+        IBlockOperation? blockStatement,
+        CancellationToken cancellationToken)
+    {
+        var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var compilation = semanticModel.Compilation;
+
+        var enumType = compilation.GetSpecialType(SpecialType.System_Enum);
+        var enumIsDefinedGenericMethod = enumType.GetMembers(nameof(Enum.IsDefined)).FirstOrDefault(m => m is IMethodSymbol { IsStatic: true, Arity: 1, Parameters.Length: 1 });
+
+        var editor = new SyntaxEditor(root, document.Project.Solution.Services);
+        var generator = editor.Generator;
+
+        var parameterIdentifierName = generator.IdentifierName(parameter.Name);
+        var typeOfParameterExpression = generator.TypeOfExpression(generator.TypeExpression(parameter.Type));
+
+        SyntaxNode enumIsDefinedInvocation;
+        if (enumIsDefinedGenericMethod is not null)
+        {
+            enumIsDefinedInvocation = generator.InvocationExpression(
+                generator.MemberAccessExpression(
+                    generator.TypeExpression(enumType),
+                    generator.GenericName(enumIsDefinedGenericMethod.Name, parameter.Type)),
+                parameterIdentifierName);
+        }
+        else
+        {
+            enumIsDefinedInvocation = generator.InvocationExpression(
+                generator.MemberAccessExpression(
+                    generator.TypeExpression(enumType),
+                    nameof(Enum.IsDefined)),
+                typeOfParameterExpression,
+                parameterIdentifierName);
+        }
+
+        var finalCondition = generator.LogicalNotExpression(enumIsDefinedInvocation);
+
+        var invalidEnumArgumentExceptionType = compilation.GetBestTypeByMetadataName(typeof(InvalidEnumArgumentException).FullName!);
+        Contract.ThrowIfNull(invalidEnumArgumentExceptionType);
```

Comment:
how do you know this is safe?  was this checked earlier during the analysis phase?

Reply:
[This type existed since .NET Framework 1.1.](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.invalidenumargumentexception?view=net-9.0) We can safely assume, that it is not null here. Again, similar logic is applied to `ArgumentNullException` type

Reply:
I think features are usually expected to be resilient against the case they are running in a project with incomplete or no references.